### ISSUE TITLE
Question: using render method for empty data

### ISSUE THREAD

**Issue by Rdx31** [EmCousin/grape-jsonapi#31]
Good afternoon !

I have a question regarding an edge case that I am experiencing using the gem.

**Context**:
I am sometimes sending to the `render` method an empty array - if there are no objects to return - , and in addition some metadata passed as options:

Example:
```ruby
response = []
metadata = {pagination: { page: 1, page_size: 2}}
render response, meta: metadata
```
I get in reply only
```
[]
```

This surprises me, because I was expecting the below reply, which seems for me consistent with the [jsonapi specification](https://jsonapi.org/format/#fetching-resources-responses-200) for empty data:
```js
{
  "data": [],
  "meta": {
    "pagination": {
      "page": 1,
      "page_size": 2
    }
  }
}
```

The `[]` answer is due to the fact that, by the implementation, `serializable_class` method will render `nil` on an empty array, thus the method `serialize_resource` will then perform a `resource.map(&:serializable_hash)`, which ultimately leads to an empty array.

To override this, and get my expected answer payload, I have to do this little trick:
```ruby
render data: response, meta: metadata
```

**Questions:**
To be sure that we both understand the same thing about the `jsonapi` specification for empty data, could you confirm that for empty array of data, the expected return of the `render` method should be:
```js
{
  "data": []
}
```

If not, how should I use the `render` method for an empty array of data and my additional options ?

**Comment by EmCousin** [comment]
Hi @Rdx31, thanks for posting this issue and sorry for the delay!

The problem is due to `jsonapi_serializable` not being able to find a serializer class that `include JSONAPI::Serializer`, therefore it falls back to calling `serializable_hash` on each object.

At first glance, one way to work around it would be to return an empty ActiveRecord collection, if you use ActiveRecord. That way it would find the corresponding JSONAPI::Serializer and return a response with the proper JSONAPI spec compliant format. Something like:
```ruby
response = User.none # returns an instance of User::ActiveRecord_Relation
metadata = {pagination: { page: 1, page_size: 2}}
render response, meta: metadata
```

Another way could be in two steps:
1. modify the `serialize_resource` method in a way that it falls back to `serialize` instead of just calling `serializable_hash` on each object. It would thus check if each item has a corresponding serializer
2. check if `resource` is an empty array and return the resource nested in a JSONAPI compliant object

```diff
-def serialize_resource(resource, options)
-  jsonapi_serializable(resource, options)&.serializable_hash || resource.map(&:serializable_hash)
-end 
+def serialize_resource(resource, env)
+  jsonapi_serializable(resource, json_api_options(env))&.serializable_hash || resource.map { |item| serialize(item, env) }
+end 
```
I don't have a lot of time at the moment, but I'll give it a go whenever I can. Or feel free to open a PR if you want to tacle it ! 🙂

**Comment by EmCousin** [comment]
@Rdx31 I opened [this PR](https://github.com/EmCousin/grape-jsonapi/pull/32) ; can you check it out and validate this covers your needs?

**Comment by Rdx31** [comment]
@EmCousin yes that covers exactly my issue !

**Comment by EmCousin** [comment]
@Rdx31 changes released on version 1.0.1 at https://rubygems.org/gems/grape-jsonapi
